Applications/AlexNet/jni/main.cpp

```diff
@@ -0,0 +1,188 @@
+#include <vector>
+
+#include <cifar_dataloader.h>
+#include <model.h>
+
+/**
+ * @brief     Data size for each category
+ */
+const unsigned int num_class = 100;
+
+const unsigned int num_train = 100;
+
+const unsigned int num_val = 20;
+
+const unsigned int batch_size = 128;
+
+const unsigned int feature_size = 3072;
+
+unsigned int train_count = 0;
+unsigned int val_count = 0;
+
+int width = 32;
+
+int height = 32;
+
+int channel = 3;
+
+unsigned int seed;
+
+std::string resource;
+
+float training_loss = 0.0;
+float validation_loss = 0.0;
+float last_batch_loss = 0.0;
+
+using ModelHandle = std::unique_ptr<ml::train::Model>;
+using UserDataType = std::unique_ptr<nntrainer::alex::DataLoader>;
+
+int trainData_cb(float **input, float **label, bool *last, void *user_data) {
+  auto data = reinterpret_cast<nntrainer::alex::DataLoader *>(user_data);
+
+  data->next(input, label, last);
+  return 0;
+}
+
+int validData_cb(float **input, float **label, bool *last, void *user_data) {
+  auto data = reinterpret_cast<nntrainer::alex::DataLoader *>(user_data);
+
+  data->next(input, label, last);
+  return 0;
+}
+
+std::array<UserDataType, 2>
+createRealDataGenerator(const std::string &directory, unsigned int batch_size,
+                        unsigned int data_split) {
+
+  UserDataType train_data(new nntrainer::alex::Cifar100DataLoader(
+    directory + "/alex_trainingSet.dat", batch_size, 1));
+  UserDataType valid_data(new nntrainer::alex::Cifar100DataLoader(
+    directory + "/alex_valSet.dat", batch_size, 1));
+
+  return {std::move(train_data), std::move(valid_data)};
+}
+
+int main(int argc, char *argv[]) {
+
+  if (argc < 3) {
+    std::cout << "./nntrainer_alex alex.ini resource\n";
+    exit(-1);
+  }
+
+  seed = time(NULL);
+  srand(seed);
+
+  const std::vector<std::string> args(argv + 1, argv + argc);
+  std::string config = args[0];
+  resource = args[1];
+
+  std::array<UserDataType, 2> user_datas;
+
+  try {
+    user_datas = createRealDataGenerator(resource, batch_size, 1);
+  } catch (const std::exception &e) {
+    std::cerr << "uncaught error while creating data generator! details: "
+              << e.what() << std::endl;
+    return 1;
+  }
+
+  auto &[train_user_data, valid_user_data] = user_datas;
+
+  std::unique_ptr<ml::train::Dataset> dataset_train;
+  try {
+    dataset_train = ml::train::createDataset(
+      ml::train::DatasetType::GENERATOR, trainData_cb, train_user_data.get());
+  } catch (const std::exception &e) {
+    std::cerr << "Error during create train dataset: " << e.what() << std::endl;
+    return 1;
+  }
+
+  std::unique_ptr<ml::train::Dataset> dataset_valid;
+  try {
+    dataset_valid = ml::train::createDataset(
+      ml::train::DatasetType::GENERATOR, validData_cb, valid_user_data.get());
+  } catch (const std::exception &e) {
+    std::cerr << "Error during create valid dataset: " << e.what() << std::endl;
+    return 1;
+  }
+
+  /**
+   * @brief     Neural Network Create & Initialization
+   */
```

Comment:
What about using a simple one instead of formal formatting as below?
```cpp
/** Neural Network Create & Initialization */
```

Reply:
It also check NNTrainer's coding convention or Discuss with other member.

Reply:
I got it.